src/Lender.sol

```diff
@@ -548,20 +519,20 @@ contract Lender {
         // Fetch the principal token for this lend call
         address pt = IMarketPlace(marketPlace).markets(u, m, p);
 
-        // Get the starting balance for the Lender
-        uint256 starting = IERC20(pt).balanceOf(address(this));
-
         // Conduct the lend operation to acquire principal tokens
-        (bool success, ) = adapter.delegatecall(
+        (bool success, bytes memory returndata) = adapter.delegatecall(
             abi.encodeWithSignature('lend(bytes calldata inputdata)', d) // TODO: create lend signature
         );
 
         if (!success) {
             revert Exception(0, 0, 0, address(0), address(0)); // TODO: assign exception
         }
 
-        // Fetch how many principal tokens were obtained
-        uint256 obtained = IERC20(pt).balanceOf(address(this)) - starting;
+        // Get the amount of PTs (in protocol decimals) received
+        uint256 obtained = abi.decode(returndata, (uint256));
+
+        // Extract fee
+        fees[u] += a / feenominator;
```

Comment:
Do we assume that the `obtained` already has the fee calculated / removed within the adapter?

Reply:
`obtained` refers to the amount of PTs received in the transaction -- I have added a second return parameter that contains the total amount of underlying spent by the user (including the fee).